### ISSUE TITLE
NPM Cache Permissions set by Sudo

### ISSUE THREAD

**Issue by ravedoglv** [hoobs-org/HOOBS#235]
upgraded to beta 70 and homebridge wont load now - missing plugins - nor can you reinstall them (WHICH I SHOULDN'T HAVE TO)

**Version**
beta 70

**Did you upgrade**
Please let us know if you upgraded from a previous version.
- [X ] Yes
- [ ] No

**What device are you using?**
Please let us know the device you are running HOOBS on.
- [ ] HOOBS Box
- [ ] HOOBS SD Card
- [X ] Custom Install

**If this is a custom install, what's your operating system?**
Please let us know what operating system and version you are using. Ex. Fedora 30 or macOS Catalina.

**List your plugins**
Please include a list of the plugins you are using. We need to be able to find it on NPM. If the plugin is no published, please include a link to the repository.

**Post your config**
This can be found in the interface, Configuration -> Advanced or you can get it via SSH `cat ~/.hoobs/etc/config.json`

* Remove any passwords from the config file. This is public.

```
POST CONFIG HERE
```

**Post your log**
You can get the log from the intercafe. This is in the Log section.
[11/8/2019, 20:50:48] HOOBS listning on port 8080.
[Homebridge] [11/8/2019, 20:50:48] Loaded config.json with 1 accessories and 3 platforms.
[Homebridge] [11/8/2019, 20:50:48] ---
[Homebridge] [11/8/2019, 20:50:49] No plugins found. See the README for information on installing plugins.
[Homebridge] [11/8/2019, 20:50:49] Loading 3 platforms...
[Homebridge] [11/8/2019, 20:50:49] Got SIGTERM, shutting down service...
[Homebridge] [11/8/2019, 20:50:49] Error: The requested platform 'Hue' was not registered by any plugin.
[Homebridge] at API.platform (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/api.js:115:19)
[Homebridge] at Server._loadPlatforms (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:330:47)
[Homebridge] at Server.run (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:77:14)
[Homebridge] at module.exports (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/cli.js:78:12)
[Homebridge] at Object.<anonymous> (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/bin/homebridge:17:22)
[Homebridge] at Module._compile (internal/modules/cjs/loader.js:956:30)
[Homebridge] at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
[Homebridge] at Module.load (internal/modules/cjs/loader.js:812:32)
[Homebridge] at Function.Module._load (internal/modules/cjs/loader.js:724:14)
[Homebridge] at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
[List Accessories] connect ECONNREFUSED 127.0.0.1:51826
[Homebridge] [11/8/2019, 20:51:34] Loaded config.json with 1 accessories and 3 platforms.
[Homebridge] [11/8/2019, 20:51:34] ---
[Homebridge] [11/8/2019, 20:51:34] No plugins found. See the README for information on installing plugins.
[Homebridge] [11/8/2019, 20:51:34] Loading 3 platforms...
[Homebridge] [11/8/2019, 20:51:34] Error: The requested platform 'Hue' was not registered by any plugin.
[Homebridge] at API.platform (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/api.js:115:19)
[Homebridge] at Server._loadPlatforms (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:330:47)
[Homebridge] at Server.run (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:77:14)
[Homebridge] at module.exports (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/cli.js:78:12)
[Homebridge] at Object.<anonymous> (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/bin/homebridge:17:22)
[Homebridge] at Module._compile (internal/modules/cjs/loader.js:956:30)
[Homebridge] at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
[Homebridge] at Module.load (internal/modules/cjs/loader.js:812:32)
[Homebridge] at Function.Module._load (internal/modules/cjs/loader.js:724:14)
[Homebridge] at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
[Homebridge] [11/8/2019, 20:51:34] Got SIGTERM, shutting down service...
[List Accessories] connect ECONNREFUSED 127.0.0.1:51826
[List Rooms] connect ECONNREFUSED 127.0.0.1:51826
[Homebridge] [11/8/2019, 20:53:58] Loaded config.json with 1 accessories and 3 platforms.
[Homebridge] [11/8/2019, 20:53:58] ---
[Homebridge] [11/8/2019, 20:53:58] No plugins found. See the README for information on installing plugins.
[Homebridge] [11/8/2019, 20:53:58] Loading 3 platforms...
[Homebridge] [11/8/2019, 20:53:58] Error: The requested platform 'Hue' was not registered by any plugin.
[Homebridge] at API.platform (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/api.js:115:19)
[Homebridge] at Server._loadPlatforms (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:330:47)
[Homebridge] at Server.run (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:77:14)
[Homebridge] at module.exports (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/cli.js:78:12)
[Homebridge] at Object.<anonymous> (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/bin/homebridge:17:22)
[Homebridge] at Module._compile (internal/modules/cjs/loader.js:956:30)
[Homebridge] at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
[Homebridge] at Module.load (internal/modules/cjs/loader.js:812:32)
[Homebridge] at Function.Module._load (internal/modules/cjs/loader.js:724:14)
[Homebridge] at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
[Homebridge] [11/8/2019, 20:53:58] Got SIGTERM, shutting down service...
[List Accessories] connect ECONNREFUSED 127.0.0.1:51826
[List Rooms] connect ECONNREFUSED 127.0.0.1:51826
[Homebridge] [11/8/2019, 20:53:58] Loaded config.json with 1 accessories and 3 platforms.
[Homebridge] [11/8/2019, 20:53:58] ---
[Homebridge] [11/8/2019, 20:53:58] No plugins found. See the README for information on installing plugins.
[Homebridge] [11/8/2019, 20:53:58] Loading 3 platforms...
[Homebridge] [11/8/2019, 20:53:58] Error: The requested platform 'Hue' was not registered by any plugin.
[Homebridge] at API.platform (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/api.js:115:19)
[Homebridge] at Server._loadPlatforms (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:330:47)
[Homebridge] at Server.run (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:77:14)
[Homebridge] at module.exports (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/cli.js:78:12)
[Homebridge] at Object.<anonymous> (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/bin/homebridge:17:22)
[Homebridge] at Module._compile (internal/modules/cjs/loader.js:956:30)
[Homebridge] at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
[Homebridge] at Module.load (internal/modules/cjs/loader.js:812:32)
[Homebridge] at Function.Module._load (internal/modules/cjs/loader.js:724:14)
[Homebridge] at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
[Homebridge] [11/8/2019, 20:53:58] Got SIGTERM, shutting down service...
[List Accessories] connect ECONNREFUSED 127.0.0.1:51826
npm
 ERR! code EACCES
npm ERR! syscall mkdir
npm
ERR!
path /home/hoobs/.npm/_cacache/index-v5/e1/a8
npm ERR!
 errno -13
npm
 ERR! 
npm ERR! Your cache folder contains root-owned files, due to a bug in
npm ERR! previous versions of npm which has since been addressed.
npm ERR! 
npm
 ERR! To permanently fix this problem, please run:
npm ERR!   sudo chown -R 1000:1000 "/home/hoobs/.npm"
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/hoobs/.npm/_logs/2019-11-09T04_57_46_943Z-debug.log
[11/8/2019, 20:57:47] Installed "homebridge-camera-ffmpeg" package.
[List Accessories] connect ECONNREFUSED 127.0.0.1:51826
[List Accessories] connect ECONNREFUSED 127.0.0.1:51826
[List Rooms] connect ECONNREFUSED 127.0.0.1:51826```
POST LOG HERE
```

**Additional context**
Add any other context about the problem here.

**Comment by Holydevil66** [comment]
Can confirm - exactly the same issue here

**Comment by mkellsy** [comment]
It looks like your .npm cache is hosed. This is a known issue with NPM.

To fix run this
sudo rm -fR ~/.npm

This will delete the NPM cache in your home folder removing the folders put there by sudo that you don't have access to.

**Comment by ravedoglv** [comment]
Question. What did your upgrade do to create this issue? All I did was notice that there was a new update then performed it using the hoobs interface. Other than that I’ve have not done anything extra in this hoobs install. Everything was done and create from with hoobs.

**Comment by ravedoglv** [comment]
ran your rm command - rebooted - still no plugins found

[11/9/2019, 11:21:30] HOOBS listning on port 8080.
[Homebridge] [11/9/2019, 11:21:31] Loaded config.json with 1 accessories and 3 platforms.
[Homebridge] [11/9/2019, 11:21:31] ---
[Homebridge] [11/9/2019, 11:21:31] No plugins found. See the README for information on installing plugins.
[Homebridge] [11/9/2019, 11:21:31] Loading 3 platforms...
[Homebridge] [11/9/2019, 11:21:31] Error: The requested platform 'Hue' was not registered by any plugin.
[Homebridge] at API.platform (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/api.js:115:19)
[Homebridge] at Server._loadPlatforms (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:330:47)
[Homebridge] at Server.run (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:77:14)
[Homebridge] at module.exports (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/cli.js:78:12)
[Homebridge] at Object.<anonymous> (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/bin/homebridge:17:22)
[Homebridge] at Module._compile (internal/modules/cjs/loader.js:956:30)
[Homebridge] at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
[Homebridge] at Module.load (internal/modules/cjs/loader.js:812:32)
[Homebridge] at Function.Module._load (internal/modules/cjs/loader.js:724:14)
[Homebridge] at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
[Homebridge] [11/9/2019, 11:21:31] Got SIGTERM, shutting down service...
[List Accessories] connect ECONNREFUSED 127.0.0.1:51826
[Homebridge] [11/9/2019, 11:21:45] Loaded config.json with 1 accessories and 3 platforms.
[Homebridge] [11/9/2019, 11:21:45] ---
[Homebridge] [11/9/2019, 11:21:45] No plugins found. See the README for information on installing plugins.
[Homebridge] [11/9/2019, 11:21:45] Loading 3 platforms...
[Homebridge] [11/9/2019, 11:21:45] Error: The requested platform 'Hue' was not registered by any plugin.
[Homebridge] at API.platform (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/api.js:115:19)
[Homebridge] at Server._loadPlatforms (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:330:47)
[Homebridge] at Server.run (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/server.js:77:14)
[Homebridge] at module.exports (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/lib/cli.js:78:12)
[Homebridge] at Object.<anonymous> (/home/hoobs/.hoobs/node_modules/@hoobs/homebridge/bin/homebridge:17:22)
[Homebridge] at Module._compile (internal/modules/cjs/loader.js:956:30)
[Homebridge] at Object.Module._extensions..js (internal/modules/cjs/loader.js:973:10)
[Homebridge] at Module.load (internal/modules/cjs/loader.js:812:32)
[Homebridge] at Function.Module._load (internal/modules/cjs/loader.js:724:14)
[Homebridge] at Function.Module.runMain (internal/modules/cjs/loader.js:1025:10)
[Homebridge] [11/9/2019, 11:21:45] Got SIGTERM, shutting down service...

**Comment by ravedoglv** [comment]
Also changing the title of this report to something that was not the original question is not helpful - since the original issue is still about missing plugins and the cache was a side issue.

**Comment by mkellsy** [comment]
I looked into the user mode initialization script. Yes a locked .npm and .node-gyp cache is causing the plugins not to install on update.

I realized that this step doesn't need to happen. Yes we are updating HOOBS, but updating the plugins is not actually what is needed. Plugin updates will need to happen from the plugin screens.

Then when a new version is detected, don't remove the user mode node_modules. This will keep the plugins in place. Yes there might be unused node modules. But there is a factory reset for that.

I am marking this fixed in 3.0.0-beta.71.
